internal/envelope/conn/weavelet_conn.go

```diff
@@ -55,28 +58,41 @@ func NewWeaveletConn(r io.ReadCloser, w io.WriteCloser, h WeaveletHandler) (*Wea
 		conn:    conn{name: "weavelet", reader: r, writer: w},
 	}
 
-	// Block until a weavelet is received.
+	// Block until the weavelet information is received.
 	msg := &protos.EnvelopeMsg{}
 	if err := d.conn.recv(msg); err != nil {
 		d.conn.cleanup(err)
 		return nil, err
 	}
-	d.wlet = msg.WeaveletInfo
-	if d.wlet == nil {
-		err := fmt.Errorf("expected WeaveletInfo, got %v", msg)
+	d.info = msg.WeaveletSetupInfo
+	if d.info == nil {
+		err := fmt.Errorf("expected WeaveletSetupInfo, got %v", msg)
 		d.conn.cleanup(err)
 		return nil, err
 	}
-	if err := runtime.CheckWeaveletInfo(d.wlet); err != nil {
+	if err := runtime.CheckWeaveletSetupInfo(d.info); err != nil {
 		d.conn.cleanup(err)
 		return nil, err
 	}
+
+	// Reply to the envelope with weavelet's runtime information.
+	lis, err := listen(d.info)
+	if err != nil {
+		d.conn.cleanup(err)
+		return nil, err
+	}
+	d.lis = lis
+	dialAddr := fmt.Sprintf("tcp://%s", lis.Addr().String())
+	info := &protos.WeaveletInfo{DialAddr: dialAddr}
+	if err := d.send(&protos.WeaveletMsg{Id: -msg.Id, WeaveletInfo: info}); err != nil {
```

Comment:
Yeah, sorry, my bad, this is leftover from when the WeaveletSetupInfo <-> WeaveletInfo was implemented as a RPC, which doesn't work because Serve() hasn't been called yet.